src/components/awards/AwardForm.tsx

```diff
@@ -57,42 +58,50 @@ export const AwardForm = ({
   onCancel,
   onChange,
 }: AwardFormProps) => {
-  const [funders, setFunders] = useState<FunderOption[]>([]) // State to hold funder options
+  const [funders, setFunders] = useState<FunderOption[]>([])
+  const [isLoading, setIsLoading] = useState(false)
+  const [isSearchFocused, setIsSearchFocused] = useState(false)
+  const [isSearchHovered, setIsSearchHovered] = useState(false)
   const formRef = useRef<FormikProps<AwardAttrs>>(null)
   const primaryButtonText = values.source ? 'Update funder' : 'Add funder'
 
-  // Fetch funders only once, caching them for future use
-  useEffect(() => {
-    const fetchFunders = async () => {
-      try {
-        const response = await fetch('https://api.crossref.org/funders')
-        if (!response.ok) {
-          throw new Error(`HTTP error! status: ${response.status}`)
-        }
-        const data = await response.json()
-        const funderOptions: FunderOption[] = data.message.items.map(
-          (funder: Funder) => ({
-            value: funder.name,
-            label: funder.name,
-          })
-        )
-        funderOptions.sort((a, b) => a.label.localeCompare(b.label))
-        // Cache the funders data
-        sessionStorage.setItem('funders', JSON.stringify(funderOptions))
-        setFunders(funderOptions)
-      } catch (error) {
-        console.error('Error fetching funders:', error)
-      }
+  const searchFunders = async (query: string) => {
+    if (!query) {
+      setFunders([])
+      return
     }
 
-    // Check if funders are already cached in sessionStorage
-    const storedFunders = sessionStorage.getItem('funders')
-    if (storedFunders) {
-      setFunders(JSON.parse(storedFunders))
-    } else {
-      fetchFunders()
+    setIsLoading(true)
+    try {
+      const response = await fetch(
+        `https://api.crossref.org/funders?query=${encodeURIComponent(query)}`
+      )
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`)
+      }
+      const data = await response.json()
+      const funderOptions: FunderOption[] = data.message.items.map(
+        (funder: Funder) => ({
+          value: funder.name,
+          label: funder.name,
+        })
+      )
+      funderOptions.sort((a, b) => a.label.localeCompare(b.label))
+      setFunders(funderOptions)
+    } catch (error) {
+      console.error('Error fetching funders:', error)
+      setFunders([])
+    } finally {
+      setIsLoading(false)
     }
-  }, []) // Only run on mount
+  }
+
+  const debouncedSearch = debounce(searchFunders, 300)
```

Comment:
This debounce would be recreated on each rerendering, wouldn't it? It probably needs a useMemo otherwise it will reset after each rerender and not work correctly all the time.